x-pack/plugins/ml/public/application/data_frame_analytics/pages/components/analytics_selector/analytics_id_selector.tsx

```diff
@@ -0,0 +1,280 @@
+      console.error('Error fetching analytics', e); // eslint-disable-line
+      toasts.addDanger({
+        title: i18n.translate('xpack.ml.analyticsSelector.analyticsFetchErrorMessage', {
+          defaultMessage: 'An error occurred fetching analytics jobs. Refresh and try again.',
+        }),
+      });
+    }
+    setIsLoading(false);
+  }
+
+  async function fetchAnalyticsModels() {
+    setIsLoading(true);
+    try {
+      const response = await trainedModelsApiService.getTrainedModels(undefined, {
+        size: 1000,
+      });
+      setTrainedModels(response);
+    } catch (e) {
+      console.error('Error fetching trained models', e); // eslint-disable-line
+      toasts.addDanger({
+        title: i18n.translate('xpack.ml.analyticsSelector.trainedModelsFetchErrorMessage', {
+          defaultMessage: 'An error occurred fetching trained models. Refresh and try again.',
+        }),
+      });
+    }
+    setIsLoading(false);
+  }
+
+  function closeFlyout() {
+    setIsFlyoutVisible(false);
+  }
+
+  // Fetch analytics jobs and models on flyout open
+  useEffect(() => {
+    fetchAnalyticsJobs();
+    if (jobsOnly === false) {
+      fetchAnalyticsModels();
+    }
+  }, []);
+
+  const applySelection: any = useCallback(() => {
+    setAnalyticsId(selected);
+    closeFlyout();
+  }, [selected?.model_id, selected?.job_id]);
+
+  const pagination = {
+    initialPageSize: 5,
+    pageSizeOptions: [3, 5, 8],
+  };
+
+  const selectionValue = {
+    selectable: (item: any) => {
+      const selectedId = selected?.job_id || selected?.model_id;
+      return selected === undefined || selectedId === item.id || selectedId === item.model_id;
+    },
+    onSelectionChange: (selectedItem: any) => {
+      const item = selectedItem[0];
+      const config = item?.analysis || item?.inference_config;
+      const analysisType = item ? Object.keys(config)[0] : undefined;
+
+      setSelected({
+        model_id: item?.model_id,
+        job_id: item?.id,
+        analysis_type: analysisType,
+      });
+    },
+  };
+
+  const tabs = [
+    {
+      id: 'Jobs',
+      name: i18n.translate('xpack.ml.analyticsSelector.jobsTab', {
+        defaultMessage: 'Jobs',
+      }),
+      content: (
+        <EuiInMemoryTable
+          items={analyticsJobs}
+          itemId="id"
+          loading={isLoading}
+          columns={columns}
+          pagination={pagination}
+          sorting={true}
+          selection={selectionValue}
+          isSelectable={true}
+        />
+      ),
+    },
+  ];
+
+  if (jobsOnly === false) {
+    tabs.push({
+      id: 'Models',
+      name: i18n.translate('xpack.ml.analyticsSelector.modelsTab', {
+        defaultMessage: 'Models',
+      }),
+      content: (
+        <EuiInMemoryTable
+          items={trainedModels}
+          itemId="model_id"
+          loading={isLoading}
+          columns={modelColumns}
+          pagination={pagination}
+          sorting={true}
+          selection={selectionValue}
```

Comment:
Is it possible to switch the selection controls from checkboxes to radio buttons, as we use for the Single Metric Viewer?

Reply:
The job selection for AD is a custom table component so I'd need to create a table similar to the one in AD or refactor that one for re-use. Might be good for a follow up.

Reply:
OK! Not a high priority, as you can't actually select more than one row.